src/main/java/evaluation/GameEvaluator.java

```diff
@@ -147,6 +148,59 @@ public double evaluate(int[] settings) {
         return tuningGame ? gameHeuristic.evaluateGame(newGame) : stateHeuristic.evaluateState(newGame.getGameState(), playerIndex);
     }
 
+    /**
+     * Method used to return more information about the game run (e.g. for map elites)
+     *
+     * The behaviours specified at the end of this method should match (in name, and type of data recorded) to those defined in:
+     * @see ParameterSearchMapElites#getSearchFramework
+     */
+    public Pair<Double, TreeMap<String, Object>> evaluateWithStats(int[] point) {
+        Object configuredThing = searchSpace.getAgent(point);
+        boolean tuningPlayer = configuredThing instanceof AbstractPlayer;
```

Comment:
I'm getting conniptions at the oodles of exactly duplicated code from evaluate()...any chance that this could extracted out to a helper function to reduce duplication?